dlrover/trainer/torch/flash_checkpoint/engine.py

```diff
@@ -335,6 +346,25 @@ def get_state_dict_from_memory(self):
             logger.info(f"Load checkpoint at step {config.step} from memory.")
         return config.step, state_dict
 
+    def _restore_memory_from_replica(self):
+        if not self._replica_manager.has_replica():
+            return
+        self._shm_handler.init_shared_memory()
+        byte_tensor, meta = self._replica_manager.gather(self._shm_handler)
+        if (
+            byte_tensor is not None
+            and meta
+            and not self._shm_handler.shared_memory
+        ):
+            shm_size = byte_tensor.size()[0]
+            self._shm_handler.init_shared_memory(create=True, size=shm_size)
+            self._shm_handler.metadata.set(meta)
+            logger.info(
+                "Restore the checkpoint from the replica in the "
```

Comment:
Better print the variables for metadata, size, etc.

Reply:
> Better print the variables for metadata, size, etc.

The variables may be too many and we can print the size in the log.